recipes/automake/all/conanfile.py

```diff
@@ -107,5 +107,7 @@ def package_info(self):
         self.output.info("Setting AUTOMAKE to {}".format(automake))
         self.env_info.AUTOMAKE = automake
 
+        self.output.info("Append M4 include directories to AUTOMAKE_CONAN_INCLUDES environment variable")
```

Comment:
Is this something the consumers need to add themselves? Why not populating here that environment variable?

Reply:
Indeed, consumers need to populate this.
pkgconf[https://github.com/conan-io/conan-center-index/pull/1013] is an example:

A default autoconf/automake installation provides a few standard macros to generate the `configure` script.
pkgconf provides a `m4` file which extends this list of macros with pkg-config related macros (see pkgconf's test_package for an example).

Please also check out the [zbar pull request](https://github.com/conan-io/conan-center-index/pull/2029#issuecomment-648994253). There we found out that the default autoconf/automake misses macros related to iconv to which I don't have a good solution yet.

This would also be a good use case of [your context concept](https://github.com/conan-io/conan-center-index/pull/1985#issuecomment-649707380). It would be nice to require a dependency, but only in the context of running `autoreconf`.